### PR TITLE
[1LP][RFR] common provider widgetastic conversion

### PR DESCRIPTION
**Pull request by izapolsk**
{{pytest: cfme/tests/cloud/test_providers.py  cfme/tests/infrastructure/test_providers.py cfme/tests/middleware/test_middleware_provider.py  cfme/tests/containers/test_ssl_providers.py }}

This PR only replaces old web_ui code.  common/Provider and the rest of providers require some refactoring but it will be done later.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 7bd363f6e20f9e65b0ca9d4da30f271f8ea83c8c:

`cfme/common/provider.py`:
- :bangbang: Line 325:21: **E126** *continuation line over-indented for hanging indent*

`cfme/containers/provider/__init__.py`:
- :red_circle: Line 116:20: **F821** *undefined name 'ProviderDetailsView'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 89bfd573cb2a76d45d28e8632cb035723b20da35:

`cfme/common/provider.py`:
- :bangbang: Line 322:21: **E126** *continuation line over-indented for hanging indent*

`cfme/containers/provider/__init__.py`:
- :red_circle: Line 116:20: **F821** *undefined name 'ProviderDetailsView'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/common/provider.py [hunk 1]
```diff
@@ -165,107 +122,75 @@ def create(self, cancel=False, validate_credentials=True, check_existing=False,
...
+                    if (validate_credentials and hasattr(endp_view, 'validate') and
+                            endp_view.validate.is_displayed):
+                        # there are some endpoints which don't demand validation like
+                        #  RSA key pair
+                        endp_view.validate.click()
+                        # Flash message widget is in add_view, not in endpoints tab
+                        logger.info(
+                            'Validating credentials flash message for endpoint %s',
+                            endpoint_name)
+                        add_view.flash.assert_no_error()
```
Comment (quarckster) [inline cfme/common/provider.py]: Not needed

File: cfme/common/provider.py [hunk 2]
```diff
@@ -165,107 +122,75 @@ def create(self, cancel=False, validate_credentials=True, check_existing=False,
...
+
+            main_view = self.create_view(self.main_view)
+            if cancel:
                 created = False
+                add_view.cancel.click()
+                cancel_text = ('Add of {} Provider was '
+                               'cancelled by the user'.format(self.string_name))
+
+                main_view.entities.flash.assert_message(cancel_text)
+                main_view.entities.flash.assert_no_error()
```
Comment (quarckster) [inline cfme/common/provider.py]:
```python
main_view.entities.flash.assert_success_message(cancel_text)
```

File: cfme/common/provider.py [hunk 3]
```diff
@@ -293,123 +218,94 @@ def update(self, updates, cancel=False, validate_credentials=True):
...
+        main_view = self.create_view(self.main_view)
 
-            # filling endpoints
+        if cancel:
+            edit_view.cancel.click()
+            cancel_text = 'Edit of {type} Provider "{name}" ' \
+                          'was cancelled by the user'.format(type=self.string_name,
+                                                             name=self.name)
+            main_view.entities.flash.assert_message(cancel_text)
+            main_view.entities.flash.assert_no_error()
```
Comment (quarckster) [inline cfme/common/provider.py]:
```python
main_view.entities.flash.assert_success_message(cancel_text)
```

File: cfme/common/provider.py (hunk 1 above)
Comment (izapolsk) [reply cfme/common/provider.py]: @quarckster, I agree but this is old code which I just shifted when removed main if statement.
I'm planning to refactor whole providers structure soon. So, I'll refactor all code related to common/provider.

**Pull request merged**